New entry for pip python (#272)
Codecademy/docs: content/python/concepts/pip/pip.md

---
Title: 'pip'

> Description: 'Much like how npm is used to manage code packages in JavaScript, pip also offers practical ways to install and use pre-made code.'

Reviewer: I think this `Description` is on the right path but I think we should complete the contrast by mentioning Python, similar to the entry's introduction:

```
Description: 'Much like how npm is a package manager for JavaScript, pip is the same community-driven manager, but for Python.'

Tags:
  - 'Conceptual'
  - 'Packages'
CatalogContent:
  - 'learn-python-3'
  - 'paths/learn-python3'
---

`pip` is the package installer used in Python to install code packages. A similar tool is Node Package Manager (npm) in JavaScript. Much like npm, `pip` is the same community driven package space, but for Python.

## Packages

> The concept of packages is not unique to `pip`. A package is essentially a collection of code that is designed for a specific purpose. One example is the [neuralart](https://pypi.org/project/neuralart/) project, which is a project on Python Package Index (PyPI) that is used for "rendering generative art from a randomly initialized neural network." It's a very cool package that creates computer generated art.

Reviewer: > A package is essentially a collection of code that is designed for a specific purpose.

In order to engage with as many readers as possible, I would avoid the use of words like: 

* Easy/easily
* Basic(ally)
* Essential(ly)
* Simple/simply

Also, while it is true that packages are designed for a specific purpose, it's important to mention that they are meant to be distributed and reused by other developers in their own projects.

Reviewer: > One example is the [neuralart](https://pypi.org/project/neuralart/) project, which is a project on Python Package Index (PyPI) that is used for "rendering generative art from a randomly initialized neural network." It's a very cool package that creates computer generated art.

I would omit the last sentence since the previous sentence already describes what the package is and what it does.

## Python Package Index Repository (PyPI)

> [The Python Package Index Repository](https://pypi.org/), otherwise called PyPI, hosts a large collection of these packages. The primary tool used to interact with PyPI is the command line, specifically `pip`. Packages on this network typically offer a list of requirements to run the package, how to install it, as well as examples of the package being implemented

Reviewer: I think the abbreviation, `PyPI`, could be moved next to the expanded equivalent and enclosed in parenthesis. LIke so,

```
The [Python Package Index Repository (or **PyPI**)](https://pypi.org/) hosts a large collection...
```

## `pip` installation

> `pip` commonly comes with the installation of Python. Therefore, the first step in installing `pip` is verifying that it is already installed. This step can vary between versions of Python as well as what machine it is running on. The examples below are for version 3.6.3 and above. They are also meant for Windows. A good resource for more specific instructions would be the [official python tutorial on installing packages](https://packaging.python.org/tutorials/installing-packages/).

Reviewer: After the release of `pip` v20.3, [support for Python2 and below is ended]( https://pip.pypa.io/en/stable/development/release-process/?highlight=python%202#python-2-support). Additionally, the language's maintainers at the Python Software Foundation announced [the sunsetting of Python 2 starting January 2020](https://www.python.org/doc/sunset-python-2/). 

Should we have a disclaimer somewhere that states this entry assumes the user/reader is using Python 3?

> ```

Reviewer: In addition to using the `py` alias for Windows, should we also include code snippets using aliases for other systems like Mac or Linux?

```
# Windows
py --version

# Unix/macOS
python3 --version
```

py --version
```

> At the time of writting this, the latest version of Python is 3.9.7. To upgrade Python, [python.org](python.org) will have the latest version for install.

Reviewer: Per my previous comment, I think this would be a great place to disclaim that Python 3 is necessary to get an ideal experience out of `pip`.

> Now, to check the latest version of `pip` the following command should be run in the terminal:

Reviewer: I'd omit "Now," and just begin the sentence as:

```
The following command is run to check the latest version `pip`:
```

```
py -m pip --version
```

> If `pip` is already installed, it's version will appear in the terminal.

Reviewer: As of Python v3.4, `pip` is installed by default when Python 3.4 is installed.

https://docs.python.org/3/installing/index.html#key-terms

---
Title: 'pip'

> Description: 'Much like how npm is used to manage code packages in JavaScript, pip also offers practical ways to install and use pre-made code, but for Python. pip can be a practical tool when doing backend development. One example is Flask, which is a package that facilitates the creation of server routes. '

Reviewer 2: ```suggestion
Description: 'pip offers practical ways to install and use pre-made code for Python.'
```

> Tags:

Reviewer 2: Missing:

```
Subjects:
  - 'Computer Science'
  - 'Data Science'
  ```

  - 'Conceptual'
  - 'Packages'
CatalogContent:
  - 'learn-python-3'
  - 'paths/learn-python3'
---

> `pip` is the package installer used in Python to install code packages. A similar tool is Node Package Manager (npm) in JavaScript. Much like npm, `pip` is the same community driven package space, but for Python.

Reviewer 2: ```suggestion
[pip](https://pypi.org/project/pip) is the package installer used in Python to install code packages. A similar tool is Node Package Manager (npm) in JavaScript. Much like npm, `pip` is the same community-driven package space, but for Python.
```

## Packages

The concept of packages is not unique to `pip`. A package is a collection of code that is designed for a specific purpose. These packages are also meant to be re-used and distributed by other developers. One example is [Flask](https://pypi.org/project/Flask/). Much like how [Express](https://expressjs.com/) can facilitate the creation of server routes for backend development, Flask can offer the same reproduceable and concise solution for developers who use Python.

## Python Package Index Repository (PyPI)

[The Python Package Index Repository](https://pypi.org/) (PyPI), hosts a large collection of packages. The official package installer of the PyPI is `pip`, which is used in the command line. A good practice to understand how to use a package is reading it's documentation on PyPI. Documentation on PyPI typically comes with the following:

- A list of requirements to run the package.
- Instructions for installing the package.
- One or more examples of how the package is implemented.

> ## `pip` installation

Reviewer: Let's capitalize "installation" since it's in the header.
